### ISSUE TITLE
Ipython console could not connecting to kernel 

### ISSUE THREAD

**Issue by somashish** [spyder-ide/spyder#3124]
```
Spyder Internal Console

This console is used to report application
internal errors and to inspect Spyder
internals with the following commands:
  spy.app, spy.window, dir(spy)

Please don't use it to run your code

>>> C:\Anaconda2\lib\site-packages\nbformat\current.py:19: UserWarning: nbformat.current is deprecated.

- use nbformat for read/write/validate public API
- use nbformat.vX directly to composing notebooks of a particular version

  """)
Traceback (most recent call last):
  File "C:\Anaconda2\lib\site-packages\spyderlib\plugins\ipythonconsole.py", line 656, in toggle_view
    self.create_new_client(give_focus=False)
  File "C:\Anaconda2\lib\site-packages\spyderlib\plugins\ipythonconsole.py", line 832, in create_new_client
    self.main.extconsole.start_ipykernel(client, give_focus=give_focus)
  File "C:\Anaconda2\lib\site-packages\spyderlib\plugins\externalconsole.py", line 1273, in start_ipykernel
    ipyclient=client, give_ipyclient_focus=give_focus)
  File "C:\Anaconda2\lib\site-packages\spyderlib\plugins\externalconsole.py", line 832, in start
    show_elapsed_time=show_elapsed_time)
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\externalshell\pythonshell.py", line 207, in __init__
    show_elapsed_time=show_elapsed_time)
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\externalshell\baseshell.py", line 73, in __init__
    self.shell = self.SHELL_CLASS(parent, get_conf_path(history_filename))
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\externalshell\pythonshell.py", line 39, in __init__
    PythonShellWidget.__init__(self, parent, history_filename, profile)
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\shell.py", line 673, in __init__
    ShellBaseWidget.__init__(self, parent, history_filename, profile)
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\shell.py", line 63, in __init__
    self.history = self.load_history()
  File "C:\Anaconda2\lib\site-packages\spyderlib\widgets\shell.py", line 508, in load_history
    encoding.writelines(rawhistory, self.history_filename)
  File "C:\Anaconda2\lib\site-packages\spyderlib\utils\encoding.py", line 206, in writelines
    return write(os.linesep.join(lines), filename, encoding, mode)
  File "C:\Anaconda2\lib\site-packages\spyderlib\utils\encoding.py", line 196, in write
    text, encoding = encode(text, encoding)
  File "C:\Anaconda2\lib\site-packages\spyderlib\utils\encoding.py", line 157, in encode
    raise RuntimeError("Incorrect encoding (%s)" % coding)
RuntimeError: Incorrect encoding (None)
Traceback (most recent call last):
  File "C:\Anaconda2\lib\site-packages\spyderlib\spyder.py", line 1875, in execute_in_external_console
    console.execute_python_code(lines)
  File "C:\Anaconda2\lib\site-packages\spyderlib\plugins\ipythonconsole.py", line 816, in execute_python_code
    client.shellwidget.execute(to_text_string(lines))
  File "C:\Anaconda2\lib\site-packages\qtconsole\history_console_widget.py", line 47, in execute
    source, hidden, interactive)
  File "C:\Anaconda2\lib\site-packages\qtconsole\console_widget.py", line 630, in execute
    self._execute(source, hidden)
  File "C:\Anaconda2\lib\site-packages\qtconsole\frontend_widget.py", line 294, in _execute
    msg_id = self.kernel_client.execute(source, hidden)
AttributeError: 'NoneType' object has no attribute 'execute'

the following are the dependencies:
IPython >=1.0      :  4.0.1 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.0 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  1.6.2 (OK)
pyflakes >=0.5.0   :  1.0.0 (OK)
pygments >=1.6     :  2.0.2 (OK)
pylint >=0.25      :  None (NOK)
qtconsole >=4.0    :  4.1.1 (OK)
rope >=0.9.2       :  0.9.4 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
sympy >=0.7.3      :  0.7.6.1 (OK)
zmq >=2.1.11       :  14.7.0 (OK)
```

**Comment by ccordoba12** [comment]
What is your Spyder version?

**Comment by somashish** [comment]
spyder 2.3.8  . Can you provide any solution...??
IT WOULD BE VERY HELPFULL...Please

**Comment by mceruso** [comment]
I have a similar issue: when launching spyder the "Ipython console" can not connect to the kernel (spinning wheel...) 

However, using the pulldown menu: "Consoles/Connect to an existing kernel" and then loading the json file (e.g. kernel-57726.json")  shown within the "Console" tab, succesfully creates a functioning ipython console.

PS: I am running spyder==2.3.9, python==2.7.11 and ipython==4.2.0.

**Comment by sdobson** [comment]
Having a similar issue with spyder == 2.3.8, python == 2.7.11 and ipython 4.1.2

**Comment by ccordoba12** [comment]
What platform are you guys working on?

**Comment by somashish** [comment]
I am working on Windows 7 64 bit

**Comment by somashish** [comment]
How would I know my json connecting file...If any one can tell me... please.

**Comment by somashish** [comment]
Thanks a lot stackoverflow....its very simple just resetting the spyder solved the problem....Finally....the rock has come back.

**Comment by alismayilov** [comment]
I was getting this error when I changed Python path to anaconda on Ubuntu 16.04. I thought may be it is anaconda path cause the problem. I changed python path to default  (, i.e. same path as a spyder) path and I still get the same error.

**Comment by alismayilov** [comment]
Sorry, I was right actually, I am getting error when I change path to Anaconda path, with default path it is working fine (after restarting spyder two times).

**Comment by ccordoba12** [comment]
This should be fixed now in Spyder 3.0.2. Please update :-)
